routes/api/upload.js

```diff
@@ -67,11 +69,16 @@ exports.upload = {
     // if minimum image dimensions are specified, make sure this is above that size:
     const quality = request.query.quality || settings.quality;
     const buffer = fs.readFileSync(filepath);
+    // see if there are image dimension requirements coming from settings or query:
+    const minSize = settings.minimumImageSize.width && settings.minimumImageSize.height ?
+      settings.minimumImageSize : {};
+    minSize.width = request.query.minwidth || minSize.width;
+    minSize.height = request.query.minheight || minSize.height;
     // skip if minimumImageSize dimensions are not set, image can be any size:
-    if (settings.minimumImageSize.width || !settings.minimumImageSize.height) {
+    if (minSize.width || !minSize.height) {
```

Comment:
this logic seems wrong - why !minSize.height?

Reply:
seems like we should have two paths, one to check if width and separate for height since they can be individually set